sessions/2019/january/session_25.md

```diff
@@ -0,0 +1,82 @@
+# SOFTWARE ENGINEERING IN JAVA
+
+## Session 25 (05/02/2019)
+
+- API specs
+- Spring alternatives
+- Engineering Design
+- GItHub project integration in Slack
+
+
+
+## Notes
+
+#### API specs - URL
+
+Roger was reviewing my homework and advising to me about API approach is not the same that web apps. In web apps we worry about returned HTML content or frontend stuff, while In API approach no don't, we are just sending JSON payloads and we aren't worrying about how data will be shown.
+
+When creating resources in RESTful API,  we POST pointing to the plural name of the resource. So, for example, if the name of the resource is `account` we'll create accounts by sending the `POST` request to the URL `/accounts`. By the other side, in web we send it to the URL `/accounts/new`.
+
+We will list accounts by sending the `GET` request to the URL `/accounts` and for specific accounts queries we send the request to the URL ` /account/:id`
+
+
+
+#### API specs - Content Type
+
+We can [retrieve different type of data][1] for web development but in API approach we just are sending and receiving data/[JSON][2].
+
+
+
+#### Spring alternatives
+
+[Micronaut Framework][3] seems very promising and its recommended for Microservices.
+
+[Struts Framework][4] isn't recommended, it is not so good as Spring and over all had a [big recent discovered vulnerability][5].  
+
+
+
+#### Engineering Design
+
+We were working on the Engineering Design for the selected requirements. We work in class the requirement to list all accounts. It was defined the programming language and framework to use.
+
+Also, were defined the HTTP endpoints behavior in the controller side, as well as the used POJOS (**Note:** this has not to be exact, even not to compile, this is just to have a general understanding).
```

Comment:
The comment about it not having to compile applies to all code you put in design, not just POJOs etc.